compiler/rustc_data_structures/src/profiling.rs

```diff
@@ -204,30 +213,41 @@ impl SelfProfilerRef {
     /// Start profiling a verbose generic activity. Profiling continues until the
     /// VerboseTimingGuard returned from this call is dropped. In addition to recording
     /// a measureme event, "verbose" generic activities also print a timing entry to
-    /// stderr if the compiler is invoked with -Ztime-passes.
+    /// stderr if the compiler is invoked with -Ztime-precise or -Ztime-passes.
     pub fn verbose_generic_activity(&self, event_label: &'static str) -> VerboseTimingGuard<'_> {
         let message =
             if self.print_verbose_generic_activities { Some(event_label.to_owned()) } else { None };
 
-        VerboseTimingGuard::start(message, self.generic_activity(event_label))
+        VerboseTimingGuard::start(
+            message,
+            self.generic_activity(event_label),
+            !self.print_extra_verbose_generic_activities,
+        )
     }
 
-    /// Like `verbose_generic_activity`, but with an extra arg.
-    pub fn verbose_generic_activity_with_arg<A>(
+    /// Start profiling an extra verbose generic activity. Profiling continues until the
+    /// VerboseTimingGuard returned from this call is dropped. In addition to recording
+    /// a measureme event, "extra verbose" generic activities also print a timing entry to
```

Comment:
```suggestion
    /// a measurement event, "extra verbose" generic activities also print a timing entry to
```

Reply:
[measureme](https://github.com/rust-lang/measureme) is the profiling library ;)